### PR TITLE
Changed appium status from `int` to `enum`

### PR DESCRIPTION
**Pull request by dpgraham**
* Added AppiumStatus enum type that maps to an int
* Had to add a Gson adapter that converts it to int because Gson doesn’t automatically convert enums to int types

### CODE REVIEW

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Http/Server.java [hunk 1]
```diff
@@ -27,7 +30,8 @@ public Server() throws IOException, ServerErrorException {
     @Override
     public Response serve(IHTTPSession session) {
         BaseResponse response = router.route(session);
-        Gson gson = new Gson();
-        return newFixedLengthResponse(response.getHttpStatus(), "application/json", gson.toJson(response.getResponse()));
+        GsonBuilder gsonBuilder = new GsonBuilder();
+        gsonBuilder.registerTypeAdapter(AppiumStatus.class, new AppiumStatusAdapter());
+        return newFixedLengthResponse(response.getHttpStatus(), "application/json", gsonBuilder.create().toJson(response.getResponse()));
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Http/Server.java]: can we use http://docs.oracle.com/javaee/6/api/javax/ws/rs/core/MediaType.html to represent media type constants?

Comment (dpgraham) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Http/Server.java]: Yeah, good call. I'll update that.

Comment (dpgraham) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Http/Server.java]: I left another todo for this one.

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Model/AppiumStatus.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * An emumeration of possible strategies.
+ */
+public enum AppiumStatus {
+    // TODO: Add the rest of the Appium Statuses
+    SUCCESS(0);
+
+    private final int statusCode;
+
+    private AppiumStatus(final int name) {
```
Comment (jlipps) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Model/AppiumStatus.java]: i find `name` a bit odd here; wouldn't it be `code` or something?

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Model/AppiumStatus.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        statusCode = name;
+    }
+
+    public int getStatusName() {
```
Comment (jlipps) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Model/AppiumStatus.java]: likewise, why do we call this a name?

**Pull request merged**